packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -383,14 +617,35 @@ export class CronjobController extends BaseController<
   }
 
   /**
-   * Handle events that should cause cronjobs to be unregistered.
+   * Handle events that could cause cronjobs to be registered
+   * and for background events to be rescheduled.
+   *
+   * @param snap - Basic Snap information.
+   */
+  private _handleSnapEnabledEvent(snap: TruncatedSnap) {
+    const events = this.getBackgroundEvents(snap.id);
+    this.rescheduleBackgroundEvents(events).catch((error) => logError(error));
+    this.register(snap.id);
+  }
+
+  /**
+   * Handle events that should cause cronjobs and background events to be unregistered.
    *
    * @param snap - Basic Snap information.
    */
   private _handleSnapUnregisterEvent(snap: TruncatedSnap) {
     this.unregister(snap.id);
   }
 
+  /**
+   * Handle events that should cause cronjobs and background events to be unregistered.
+   *
+   * @param snap - Basic Snap information.
+   */
+  private _handleSnapDisabledEvent(snap: TruncatedSnap) {
+    this.unregister(snap.id, true);
```

Comment:
I'm not following the logic here, `unregister` is called already for Snaps that are disabled or uninstalled. Why do we need a separate handler and this boolean flag?

Reply:
Because we don't want to remove background event state when disabling, a snap can become enabled again and have unexpired events.

Reply:
Why not? A snap might schedule important notifcation A and regularly send notification B via cronjob. Say the user wants to disable the snap for a bit and then re-enable it after a while, if that event was scheduled before the disabling, it is lost.

Reply:
Snaps that are disabling might be misbehaving or be malicious, generally I don't think we need to give them much special treatment. I don't have a strong opinion on this specific issue, but was just thinking that this was an area to simplify in the PR

Reply:
I'm going to keep as is, I'd rather be safer.